ui/page/main_page.go

```diff
@@ -666,3 +672,150 @@ func (mp *MainPage) LayoutTopBar(gtx layout.Context) layout.Dimensions {
 		}),
 	)
 }
+
+// postDdesktopNotification posts notifications to the desktop.
+func (mp *MainPage) postDesktopNotification(notifier interface{}) {
+	proposalNotification := mp.WL.Wallet.ReadBoolConfigValueForKey(load.ProposalNotificationConfigKey)
+	var notification string
+	switch t := notifier.(type) {
+	case wallet.NewTransaction:
+
+		switch t.Transaction.Type {
+		case dcrlibwallet.TxTypeRegular:
+			if t.Transaction.Direction != dcrlibwallet.TxDirectionReceived {
+				return
+			}
+			// remove trailing zeros from amount and convert to string
+			amount := strconv.FormatFloat(dcrlibwallet.AmountCoin(t.Transaction.Amount), 'f', -1, 64)
+			notification = fmt.Sprintf("You have received %s DCR", amount)
+		case dcrlibwallet.TxTypeVote:
+			reward := strconv.FormatFloat(dcrlibwallet.AmountCoin(t.Transaction.VoteReward), 'f', -1, 64)
+			notification = fmt.Sprintf("A ticket just voted\nVote reward: %s DCR", reward)
+		case dcrlibwallet.TxTypeRevocation:
+			notification = "A ticket was revoked"
+		default:
+			return
+		}
+
+		if mp.WL.MultiWallet.OpenedWalletsCount() > 1 {
+			wallet := mp.WL.MultiWallet.WalletWithID(t.Transaction.WalletID)
+			if wallet == nil {
+				return
+			}
+
+			notification = fmt.Sprintf("[%s] %s", wallet.Name, notification)
+		}
+
+		initializeBeepNotification(notification)
+	case wallet.Proposal:
+		switch {
+		case t.ProposalStatus == wallet.NewProposalFound:
+			notification = fmt.Sprintf("A new proposal has been added Token: %s", t.Proposal.Token)
+		case t.ProposalStatus == wallet.VoteStarted:
+			notification = fmt.Sprintf("Voting has started for proposal with Token: %s", t.Proposal.Token)
+		case t.ProposalStatus == wallet.VoteFinished:
+			notification = fmt.Sprintf("Voting has ended for proposal with Token: %s", t.Proposal.Token)
+		default:
+			notification = fmt.Sprintf("New update for proposal with Token: %s", t.Proposal.Token)
+		}
+		if proposalNotification {
+			initializeBeepNotification(notification)
+		}
+	}
+}
+
+func initializeBeepNotification(n string) {
+	absoluteWdPath, err := GetAbsolutePath()
+	if err != nil {
+		log.Error(err.Error())
+	}
+
+	err = beeep.Notify("Decred Godcr Wallet", n, filepath.Join(absoluteWdPath, "ui/assets/decredicons/qrcodeSymbol.png"))
+	if err != nil {
+		log.Info("could not initiate desktop notification, reason:", err.Error())
+	}
+}
+
+// listenForSyncNotifications starts a goroutine to watch for sync updates
+// and update the UI accordingly.
+func (mp *MainPage) listenForNotifications() {
+	// Setup listeners
+	if mp.SyncProgressListener == nil {
+		mp.SyncProgressListener = listeners.NewSyncProgress()
+	}
```

Comment:
If ANY of the listeners is non-nil, abort. Always recreate the listeners; which also mean you should always set them to nil after closing their channels.